From 6a89bdc3f8df83532faafca4d3c14b36305f900e Mon Sep 17 00:00:00 2001
From: turtton <top.gear7509@turtton.net>
Date: Thu, 27 Mar 2025 11:42:39 +0900
Subject: [PATCH] fix: Append additional library path for linux

Signed-off-by: turtton <top.gear7509@turtton.net>
---
 src/Beutl.Extensions.FFmpeg/FFmpegPath.cs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Beutl.Extensions.FFmpeg/FFmpegPath.cs b/src/Beutl.Extensions.FFmpeg/FFmpegPath.cs
index fe496bc87..c91e7ac9a 100644
--- a/src/Beutl.Extensions.FFmpeg/FFmpegPath.cs
+++ b/src/Beutl.Extensions.FFmpeg/FFmpegPath.cs
@@ -155,6 +155,8 @@ public static class FFmpegLoader
         else if (OperatingSystem.IsLinux())
         {
             paths.Add($"/usr/lib/{(Environment.Is64BitProcess ? "x86_64" : "x86")}-linux-gnu");
+            var libraryPath = Environment.GetEnvironmentVariable("LD_LIBRARY_PATH")?.Split(Path.PathSeparator) ?? [];
+            paths.AddRange(libraryPath);
         }
         else if (OperatingSystem.IsMacOS())
         {
-- 
2.48.1

